Modules/PowerShellHelp.psm1

```diff
@@ -0,0 +1,113 @@
+<#
+.SYNOPSIS
+
+New-DscResourcePowerShellHelp generates PowerShell compatable help files for a DSC
+resource module
+
+.DESCRIPTION
+
+The New-DscResourcePowerShellHelp cmdlet will review all of the MOF based resources
+in a specified module directory and will inject PowerShell help files for each resource
+in to the specified directory. These help files include details on the property types for
+each resource, as well as a text description and examples where they exist.
+
+.PARAMETER OutputPath
+
+Where should the files be saved to (ususally the "en-US" folder inside the module for US english)
+
+.PARAMETER ModulePath
+
+The path to the root of the DSC resource module (where the PSD1 file is found, not the folder for
+and individual DSC resource)
+
+.EXAMPLE
+
+This example shows how to generate help for a specific module
+
+    New-DscResourcePowerShellHelp -ModulePath C:\repos\SharePointdsc -OutputPath C:\repos\SharePointDsc\en-US
+
+#>
+function New-DscResourcePowerShellHelp
+{
+    param
+    (
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $OutputPath,
+
+        [parameter(Mandatory = $true)]
+        [System.String]
+        $ModulePath
+    )
+
+    Import-Module (Join-Path -Path $PSScriptRoot -ChildPath "MofHelper.psm1")
+
+    $mofSearchPath = (Join-Path -Path $ModulePath -ChildPath "\**\*.schema.mof")
+    $mofSchemas = Get-ChildItem -Path $mofSearchPath -Recurse
+    $mofSchemas | ForEach-Object {
+        $mofFileObject = $_
+
+        $descriptionPath = Join-Path -Path $_.DirectoryName -ChildPath "readme.md"
+        if (Test-Path -Path $descriptionPath)
+        {
+            $result = (Get-MofSchemaObject -FileName $_.FullName) | Where-Object {
+                ($_.ClassName -eq $mofFileObject.Name.Replace(".schema.mof", "")) `
+                    -and ($null -ne $_.FriendlyName)
+            }
+            Write-Verbose -Message "Generating help document for $($result.FriendlyName)"
```

Comment:
I added [cmdletbinding()] to all these functions instead - this is because the functions are being called with the -Verbose flag in AppVeyor.psm1 so should be outputting this info anyway.